pkg/messaging/sns.go

```diff
@@ -6,47 +6,44 @@ import (
 	_ "github.com/aws/aws-sdk-go/aws/credentials/stscreds"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/sns"
+	"strings"
 )
 
-// NewSNS initializes a new AWS SNS client
-func NewSNS(config *Config) (*sns.SNS, error) {
+// NewSNS initializes a new AWS SNS client and returns topicArn - default for EmailTopic topic
+func NewSNS(config *Config, params ...string) (*sns.SNS, string, error) {
 	c, err := mergeAndPopulateConfig(config)
```

Comment:
after reading this, it seems like this should be split up into two methods... one to setup the client, and the second to initialize the topic

Reply:
the client can make use of both, bu instantiating the client, then calling initialize with the correct topicName

Reply:
The original thought - any client who needs to setup the sns client will need the topicArn of a particular topic (unless this client wants to create a brand new topic).  That's why we setup sns and return the topicArn in the same method.

But yes it will be more flexible for the consuming client to use two methods - one to setup sns and the 2nd one to return topicArn.

Reply:
@mykelswitzer Done named returns